value-processor/src/org/immutables/value/processor/Immutables.generator

```diff
@@ -702,26 +702,52 @@ public [type.typeAbstract.relative] [v.names.with]([v.atNullability][v.type] val
 
   /**
    * Construct a new immutable {@code [type.name]} instance.
-[for v in type.constructorArguments]
+  [for v in type.constructorArguments]
    * @param [v.name] The value for the {@code [v.name]} attribute
-[/for]
+  [/for]
    * @return An immutable [type.name] instance
    */
   public static [type.generics.spaceAfter][type.typeValue] [type.names.of]([for v in type.constructorArguments][if not for.first], [/if][v.atNullability][eachLine v.constructorParameterInjectedAnnotations][v.type] [v.name][/for]) {
     return [type.names.of]([for v in type.constructorArguments][if not for.first], [/if][if v.requiresAlternativeStrictConstructor]([constructorAcceptType v]) [/if][v.name][/for]);
   }
+
+  [if type.hasOptionalConstructorArguments]
+  /**
+   * Construct a new immutable {@code [type.name]} instance.
+  [for v in type.constructorArguments]
+   * @param [v.name] The value for the {@code [v.name]} attribute
+  [/for]
+   * @return An immutable [type.name] instance
+   */
+  public static [type.generics.spaceAfter][type.typeValue] [type.names.of]([for v in type.constructorArguments][if not for.first], [/if][v.atNullability][eachLine v.constructorParameterInjectedAnnotations][if v.isOptionalType][unwrappedOptionalType v][else][v.type][/if] [v.name][/for]) {
+    return [type.names.of]([for v in type.constructorArguments][if not for.first], [/if][if v.requiresAlternativeStrictConstructor]([constructorAcceptType v]) [/if][if v.isOptionalType andnot v.isJdkOptional][optionalOf v]([v.name])[else if v.isJdkOptional and type.kind.isFactory][optionalOf v]([v.name])[else][v.name][/if][/for]);
+  }
+  [/if]
   [/if]
 
   /**
    * Construct a new immutable {@code [type.name]} instance.
-[for v in type.constructorArguments]
+  [for v in type.constructorArguments]
    * @param [v.name] The value for the {@code [v.name]} attribute
-[/for]
+  [/for]
    * @return An immutable [type.name] instance
    */
   public static [type.generics.spaceAfter][type.typeValue] [type.names.of]([for v in type.constructorArguments][if not for.first], [/if][v.atNullability][eachLine v.constructorParameterInjectedAnnotations][constructorAcceptType v] [v.name][/for]) {
     return [validated type false]new [type.typeImmutable.relativeRaw][type.generics.diamond]([for v in type.constructorArguments][if not for.first], [/if][v.name][/for])[/validated];
   }
+
```

Comment:
Here and in other places, we need to refine so the newline appears inside the guarding `if` directive. The reason for this is that when if statement is not executing (having false condition), we will not leave extra unnecessary newlines in generated code. As generated code prettiness is valuable for us, we follow the convention: always put newline _before_ generated methods or nested class, but _inside_ guarding `if` or `for` or `template` directives, like this

```java
[if condition]

/**
 * Method
 */
void method() {
  return;
}
[/if]
[if condition andnot condition2]

interface NestedAndUseless {}
[/if]
```

Reply:
Hm...ok I didn't realise that this was the idea behind it, I'll change it accordingly.